### ISSUE TITLE
SFTP does not start

### ISSUE THREAD

**Issue by kmm2** [wolpi/prim-ftpd#18]
I tried v3.0 from f-droid.org on Android 4.4.2 and only FTP starts.
The OS is rooted and other services like Samba work fine.
I only need sftp, can you please fix it?

**Comment by xphnx** [comment]
Same for me on Cyanogenmod 12

**Comment by varyoo** [comment]
Same here

**Comment by wolpi** [comment]
I need more information to help you with this:
1. What are your settings (except password) ?
2. Can you provide me logs ?

Maybe I can create a version tomorrow to make it easier to obtain logs.

**Comment by wolpi** [comment]
I released and uploaded a new version today which provides the option to create a log file. Usually it takes some days until it is available in app shops. When you get the new version please do this:
1. enable logging to log file
2. reproduce the error
3. send me the log
4. disable logging again

**Comment by kmm2** [comment]
OK here are screenshots and a log http://www.mediafire.com/download/sh1bvw9u6crv5m3/Archive.zip

**Comment by wolpi** [comment]
The screenshots showed me what is wrong.
You have to generate a key first!
There is a small "reload" icon. Press that to (re-)genearte a key.

I guess I should improve the UI on that ;)

**Comment by varyoo** [comment]
On my side, I think I have the keys generated... as soon as i receive the update through F-Droid, I will post my logs, thanks.

**Comment by kmm2** [comment]
"You have to generate a key first!"
Yes this was the problem :)

**Comment by varyoo** [comment]
I'm being a little late, sorry for this :-1: 
I have uninstalled the application, reinstalled, generated the key, enabled the logs, set server mode to "SFTP only", then launched the server, and it crashed, I'm running it on CyanogenMod 12
Here is my logs: https://d.maxfile.ro/zmblqshjut.csv
Thanks.

**Comment by xphnx** [comment]
The apps crashes for me too. Cyanogenmod 12.1. 
Here my log: https://pad.elbinario.net/p/sftpd_log
Regards

**Comment by wolpi** [comment]
> PD: Now I see in the log it could be related to an icon issue with CM Themes

Yes it seems that could be the issue. I will investigate that further.

**Comment by varyoo** [comment]
When I disable this theme, your app works fine.

**Comment by wolpi** [comment]
> I guess I should improve the UI on that ;)

I changed the behavior when user tries to start sftp without key. Now the key generation dialog pops up and server start is delayed.


> When I disable this theme, your app works fine.

So I did the change that is required by that cyanogen theme and added the large icon. As the app has just one icon i'm using it as small and large one. IMHO that makes not much sense so I omitted the large one in the first place.

As far as I understand the docs the large icon is optional. So I would consider it a bug in the theme when it is not able to deal with that. But I changed it anyway.

I will do a release in a few days. I want to wait for a response for issue #19.
